### PR TITLE
Drop mod_env for prettyURLs (no index.php)

### PR DESCRIPTION
**Pull request by eppfel**
In an issue at owncloud/core#24972 I proposed to drop mod_env for prettyURLs, because currently mod_rewrite has to be activated manually anyway. I replaced all occurrences of 'front_controller_active' with checking the system config 'htaccess.RewriteBase' directly.

I tested the changes manually on my live installation in 9.0.53 and 10.0.0 for some time and without problems. I could not complete a full autotest due to environment limits, but completed the corresponding unit tests and phantomJS, which is reflected in a code change and a change of one test.
@LukasReschke : Maybe you should have a look, as you introduced prettyURLs in owncloud/core#14081

I can create a PR for stable10, if desired.

As this is my first PR, any feedback on my code or procedure is appreciated.

### CODE REVIEW

File: lib/private/URLGenerator.php [hunk 1]
```diff
@@ -93,7 +94,7 @@ public function linkToRouteAbsolute($routeName, $arguments = array()) {
 	 * Returns a url to the given app and file.
 	 */
 	public function linkTo( $app, $file, $args = array() ) {
-		$frontControllerActive = (getenv('front_controller_active') === 'true');
+		$frontControllerActive = !empty($this->config->getSystemValue('htaccess.RewriteBase'));
```
Comment (eppfel) [inline lib/private/URLGenerator.php]: Did a mistake with using `empty()`, will fix it, but I had to check for null as well here.

**Comment by LukasReschke** [comment]
While I can see this being an advantage with stuff like suexec, I'm quite reluctant on adding this.

The main reason for this environment trick is to ensure that mod_rewrite is in fact really running and the Nextcloud ruleset is in place. With this change this is not more the case. So let's say somebody updates Nextcloud but has the .htaccess for some reason not writable anymore => They will wonder what's broken here.

IMO we need an approach here that is less error prone.

**Comment by eppfel** [comment]
I'm with you. So maybe a fallback solution to force using mod_rewrite and ignore mod_env? E.g. with another system config 'htaccess.forceRewrite'?
Or maybe there could be a check if the rewrite is working, when you set 'htacces.RewriteBase' or do an upgrade?

**Comment by eppfel** [comment]
@LukasReschke So, what you think about my proposals? ⬆️

**Comment by LukasReschke** [comment]
I need to think about that some more. Sorry :-)

**Comment by eppfel** [comment]
closed in favor of #1602

**Pull request closed without merging**